### PR TITLE
Fix: policy disregards tests

### PR DESCRIPTION
**Pull request by colinbruce**
## What

The policy disregards date switch-over missed a few things this PR addresses them:

* cucumber tests, it reverts some placeholder code that checked the date before executing
* cucumber tests, update the final features that, now the policy-disregards path is standard, began to fail
* rspec tests, One branch of the other_assets flow was mishandled and, following the date switch, was not covered causing the coverage to drop by 0.5% 

## Checklist

Before you ask people to review this PR:

- [ ] Tests and rubocop should be passing: `bundle exec rake`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [x] There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- [x] The PR description should say what you changed and why ~, with a link to the JIRA story.~
- [x] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: spec/requests/providers/other_assets_controller_spec.rb [hunk 1]
```diff
@@ -204,7 +204,22 @@
...
+              context 'when the calculation date is prior to the policy disregards date' do
+                let(:application) do
+                  create :legal_aid_application,
+                         :with_positive_benefit_check_result,
+                         used_delegated_functions: true,
+                         used_delegated_functions_on: Date.new(2021, 1, 5)
+                end
+                let(:policy_disregards) { false }
+
+                it 'redirects to policy disregards' do
```
Comment (mpw5) [inline spec/requests/providers/other_assets_controller_spec.rb:217]: Should this be 'redirects to check your answers'?

**Review by mpw5 — APPROVED** [review]
Looks good. One question about a spec description

File: spec/requests/providers/other_assets_controller_spec.rb (hunk 1 above)
Comment (colinbruce) [reply spec/requests/providers/other_assets_controller_spec.rb:217]: I don't think so, checking answers is 165-188
This is when they select nothing on their first runthrough and have a delegated function date before the cut-over
(mocked with `receive(:capture_policy_disregards?).and_return(false)`

Comment (mpw5) [reply spec/requests/providers/other_assets_controller_spec.rb:217]: Ah, OK. I just noticed that the test checks for redirection to redirect_to(providers_legal_aid_application_check_passported_answers_path(application)), so it seemed like there was a bit of a mismatch between the description and the assertion.

**Pull request merged**